ui/framework/src/ui-framework/UiFramework.ts

```diff
@@ -150,12 +149,12 @@ export class UiFramework {
    * @param store The single Redux store created by the host application. If this is `undefined` then it is assumed that the [[StateManager]] is being used to provide the Redux store.
    * @param i18n The internationalization service created by the application. Defaults to IModelApp.i18n.
    * @param frameworkStateKey The name of the key used by the app when adding the UiFramework state into the Redux store. If not defined "frameworkState" is assumed. This value is ignored if [[StateManager]] is being used. The StateManager use "frameworkState".
-   * @param projectServices Optional app defined projectServices. If not specified DefaultProjectServices will be used.
+   * @param iTwinAccessService Optional app defined ITwinAccess. If not specified ITwinAccessClient will be used.

```

Comment:
With us removing the `ProjectServices` implementation do we even need to have a iTwinAccessSevice on UI framework?

This is the only part in all of iTwin.js that is actually using an implementation of a client. Everywhere else it really seems to be the contextId is passed around as a simple string just to be able to make all of the service calls we need with it.

Given that, I've been debating while reading through this PR if we should have it on IModelApp/IModelHost, my thinking has been no but then it does mean that UIFramework would need it (if it is actually using it).

If it turns out it's not necessarily needed here, I'd be inclined for us to slim down what we have in iTwin.js Core (this repo) in regards to the Interface you've created. Then use your interface as the starting point for both the new client lib and the mock used by tests.

Reply:
I do think you are onto something and that it may be best to remove `iTwinAccessService` as well. Currently within this monorepo it is only being used in some `test-apps` which have already had much of their functionality removed such as the now deprecated Favorites, Recently Used, and Invited properties.

Beyond the calls made through `iTwinAccessService` the interface is used:
- once in a `test-app` to retrieve an `ITwin`'s name by id exclusively for logging purposes
- once by `ExtensionServiceTool` to get an `ITwin`'s id via the its name
- eleven times in tests to get an `ITwin`'s id via the its name

This leads me to be skeptical about the necessity of this interface, and if it is being used correctly to begin with.

Reply:
I have gone through and removed `iTwinAccessService` from the `UI-Framework`. The projects build and run tests successfully, but there is a chance several of the `test-apps` no longer work and should be investigated before completing this PR.

Reply:
Are the test apps working and good to go?

Reply:
Trying them out right now, I hit an issue with the ui-test-app having an infinite spinner when trying to load the projects. Digging into it now to see what's going on.

Reply:
Marking this as resolved since this open conversation is a duplicate of a later conversation.